dask/tests/test_order.py

```diff
@@ -1,10 +1,15 @@
-from itertools import chain
+import random
 
-from dask.order import child_max, ndependents, order
+import pytest
+
+from dask.order import ndependencies, order
 from dask.core import get_deps
 from dask.utils_test import add, inc
 
-a, b, c, d, e = 'abcde'
+if random.random() < 0.5:
+    a, b, c, d, e = 'abcde'
+else:
+    a, b, c, d, e = 'abcde'[::-1]
```

Comment:
This seems dangerous to me - tests that depend on random behavior are harder to reproduce on failure, and harder to debug.

Also, having `a == 'a'` sometimes, and other times `a == 'e'` is confusing as a reader.

A few ideas:
- Always assign values in reverse order - since we sort by key names, the `order` function must be correct based on other factors.
- Build test graphs in a way that we know sorting by keyname alone won't pass the test.

Reply:
How would you feel about making the order switching explicit with `pytest.parametrize`?  I don't trust our ability to craft the graphs such that keyname won't matter.  I want to ensure that things work both ways regardless of future changes.

Reply:
Sounds reasonable to me.